### DISCUSSION TITLE
Setting up time-sensitive buffers

### DISCUSSION THREAD

**Discussion by fatLittleDog** [frePPLe/frepple#395] [Q&A] [answered]
I am trying to set up a food manufacturing process (baking) where the process is continuous, but has some natural buffers along the way.  I am wondering if I should set this up as a routing operation or as a series of fixed time/time per operations with buffers in between.

I initially set the entire process up as a routing operation with all the steps and resources along the way.  But the system seems to break everything down to the smallest quantity of the bottleneck resource - in our case the oven.  So instead of using our other resources at their max efficient capacities (eg. 1 mixer doing a full mix) it requests that the mixers do more smaller mixes (eg. same mixer doing twice as many 1/2 mixes).

I think using buffers would solve this issue, but my concern is that products not spend too much time in the buffers.  For examples, let's say that dough can spend between 20 and 30 minutes waiting to be processed.  And made pieces can spend 3.5 to 4.5 hours in the proof boxes.  And baked pieces can spend between 1 and 3 hours in the cooling area.  This allows us to put product to constantly feed the bottleneck, but the product can't sit in the buffers indefinitely.  At at certain point, the product is no longer viable.

Any ideas on how best to set up a process like this?  I am guessing there are other food processes that require the process to be continuous (once you start mixing dough you are on a path to the ovens within a certain timeframe) but also needs some flexibility in the manufacturing process.

Thanks for any thoughts!

Peter

**Comment by jdetaeye** [comment, marked as answer]
A *minimum time* in a buffer can be modeled by using the "offset" field in the operationmaterial table.  With that field you can specify that the material is only available X hours after the manufacturing has finished.

The solver doesn't have functionality to explicitly model and constrain a *maximum time* in a buffer.
In implementation projects where inventory can expire we have used a combination of:
- setting the max_early field of resources to avoid creating too-wide-gaps between operations. This is probably the most appropriate for your use case.
- some (specific and non-reusable) customizations to further enforce the previous point
- reports that show the user the material that is about to expire and allow the user to take decisions accordingly
- preprocessing planning passes to detect expiring material and automatically take some action with it

**Reply by fatLittleDog** [reply]
Thank you for the ideas - I will look at those options.
